### PR TITLE
allow color API to enable and disable colors

### PR DESCRIPTION
**Pull request by bollu**
This is useful for front-ends which wish to selectively enable or
disable coloring.

Also expose this API through the C API

### CODE REVIEW

File: src/binaryen-c.cpp [hunk 1]
```diff
@@ -3824,6 +3825,10 @@ BinaryenGetFunctionTypeBySignature(BinaryenModuleRef module,
   return NULL;
 }
 
+void BinaryenSetColosrEnabled(int enabled) { Colors::setEnabled(enabled); }
```
Comment (kripken) [inline src/binaryen-c.cpp]: typo in `Colosr`

File: src/binaryen-c.h [hunk 2]
```diff
@@ -1221,6 +1221,11 @@ BinaryenGetFunctionTypeBySignature(BinaryenModuleRef module,
                                    BinaryenType* paramTypes,
                                    BinaryenIndex numParams);
 
+// Enable or disable coloring for the WASM printer
+void BinaryenSetColorsEnabled(int enabled);
+
+// Query whether color is enable for the WASM printer
+int BinaryenIsColorsEnabled();
```
Comment (kripken) [inline src/binaryen-c.h]: I think grammatically this should be `AreColosEnabled`

File: src/support/colors.h [hunk 3]
```diff
@@ -20,6 +20,8 @@
 #include <iosfwd>
 
 namespace Colors {
+void setEnabled(bool enabled);
+bool isEnabled();
 void disable();
```
Comment (kripken) [inline src/support/colors.h]: I think we should remove the `disable()` API, and replace all callers with calls to `setEnabled(false)`.

**Comment by kripken** [comment]
Have you joined the w3c group (which we require from contributors)?

For a test, you can add a program in `test/example/`, like perhaps inside the kitchen sink example there which tests most of the C API. We can't actually check for colors working, but at least that the API exists I guess.

**Comment by bollu** [comment]
Yes, I just joined the WebAssembly community group: https://www.w3.org/community/webassembly/participants

**Comment by kripken** [comment]
Great, thanks!

**Pull request merged**